M/MLX/build_tarballs.jl

```diff
@@ -0,0 +1,117 @@
+    DirectorySource("./bundled"),
+]
+
+script = raw"""
+apk del cmake # Need CMake >= 3.24
+
+if [[ "$target" == *-apple-darwin* ]]; then
+    apple_sdk_root=$WORKSPACE/srcdir/MacOSX14.0.sdk
+    sed -i "s!/opt/$bb_target/$bb_target/sys-root!$apple_sdk_root!" $CMAKE_TARGET_TOOLCHAIN
+    sed -i "s!/opt/$bb_target/$bb_target/sys-root!$apple_sdk_root!" /opt/bin/$bb_full_target/$target-clang++
+fi
+
+cd $WORKSPACE/srcdir/mlx
+
+atomic_patch -p1 ../patches/cmake_system_processor-arm64-aarch64.patch
+atomic_patch -p1 ../patches/cmake-x86_64-apple-darwin.patch
+
+if [[ "$target" == *-w64-mingw32* ]]; then
+    atomic_patch -p1 ../patches/cmake-w64-mingw32.patch
+fi
+
+if [[ "$target" == *-linux-musl* ||
+      "$target" == *-w64-mingw32* ]]; then
+    atomic_patch -p1 ../patches/musl.patch
+fi
+
+CMAKE_EXTRA_OPTIONS=()
+if [[ "$target" == x86_64-apple-darwin* ]]; then
+    CMAKE_EXTRA_OPTIONS+=("-DMLX_ENABLE_X64_MAC=ON")
+    export MACOSX_DEPLOYMENT_TARGET=13.3
+elif [[ "$target" == *-freebsd* ||
+        "$target" == *-w64-mingw32* ]]; then
+    CMAKE_EXTRA_OPTIONS+=(
+        "-DMLX_BUILD_GGUF=OFF" # Disabled gguf, due to `gguflib-src/gguflib.c:4:10: fatal error: sys/mman.h: No such file or directory`
+        "-DMLX_BUILD_SAFETENSORS=OFF" # Disabled safetensors, due to `mlx/io/safetensors.cpp.obj:safetensors.cpp:(.rdata$.refptr._ZTVN3mlx4core2io18ParallelFileReaderE[.refptr._ZTVN3mlx4core2io18ParallelFileReaderE]+0x0): undefined reference to `vtable for mlx::core::io::ParallelFileReader'`
+    )
+fi
+
+libblastrampoline_target=$(echo $bb_full_target | cut -d- -f 1-3)
+if [[ "$target" != *-apple-darwin* &&
+      "$libblastrampoline_target" != armv6l-linux-* &&
+      "$bb_full_target" != i686-linux-gnu-cxx11 ]]; then
+    CMAKE_EXTRA_OPTIONS+=(
+        "-DBLAS_INCLUDE_DIRS=$includedir/libblastrampoline/LP64/$libblastrampoline_target"
+        "-DLAPACK_INCLUDE_DIRS=$includedir/libblastrampoline/LP64/$libblastrampoline_target"
+    )
+fi
+
+install_license LICENSE
+
+if [[ "$target" != aarch64-apple-darwin* ]]; then
+    cmake \
+        -B build \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=$prefix \
+        -DCMAKE_TOOLCHAIN_FILE=$CMAKE_TARGET_TOOLCHAIN \
+        -DMLX_BUILD_TESTS=OFF \
+        -DMLX_BUILD_EXAMPLES=OFF \
+        -DMLX_BUILD_METAL=OFF \
+        -DBUILD_SHARED_LIBS=ON \
+        -G Ninja \
+        ${CMAKE_EXTRA_OPTIONS[@]}
+    cmake --build build --parallel $nproc
+    cmake --install build
+    cmake --install build --component headers
+else
+    cd $WORKSPACE/srcdir
+    unzip -d mlx-$target mlx-$target.whl
+    cd mlx-$target/mlx
+    find lib -type f -name "*.$dlext" | xargs -Isrc install -D -m 755 -v src $prefix/src
+    find lib -type f -name "*.metallib" | xargs -Isrc install -D -m 644 -v src $prefix/src
```

Comment:
```suggestion
    find lib -type f \( -name "*.$dlext" -o -name "*.metallib" \) -exec install -D -m 755 -v '{}' -t "${prefix}" \;
```
should be shorter, no?

Reply:
The directory structure under `lib` (and more importantly, in `include` and `share`) must be maintained when copied to `$prefix`, so `find lib -type f ... -exec install -D ... {} -t $prefix` would not work, as the files would all end up in `$prefix`, and not `$prefix/[relative_path]`.

The `xargs` approach works, an so does: `find lib -type f ... -exec install -D -m 755 -v {} $prefix/{} \;` (without the `-t`).

The latter (`find ... -exec`) is however a little brittle - if `$prefix/{}` already exists as a directory, e.g.:
```

Reply:
Applied the `find lib -type f ... -exec install -D {} $prefix/{} \;` variant.

Reply:
Ok, yeah, sorry, I missed they had different permissions.